lib/view_component/base.rb

```diff
@@ -518,6 +519,18 @@ def inherited(child)
         # `compile` defines
         compile
 
+        child.include ActiveSupport::Configurable
+
+        if child.superclass == ViewComponent::Base
+          child.define_singleton_method(:config) do
+            @@config ||= Rails.application.config.view_component.inheritable_copy
```

Comment:
I'll leave a comment to this effect, but ViewComponent::Base starts out with the defaults. These overrides allow subclasses to inherit from Rails.application.config to start with and cascade down from there.

To achieve our goal of making sure that changes to config don't broadly interfere with gems that want to integrate with ViewComponent, we should probably make it easier here for folks to override `config` with a base config.

Reply:
I guess this'll be done by gems having their own `ApplicationComponent` equivalent now that directly inherits from `Base`.